src/OpenIddict.Client.WebIntegration/OpenIddictClientWebIntegrationHandlers.Discovery.cs

```diff
@@ -144,7 +144,7 @@ public ValueTask HandleAsync(HandleConfigurationResponseContext context)
                     context.Configuration.GrantTypesSupported.Add(GrantTypes.Implicit);
                 }
 
-                else if (context.Registration.ProviderType is ProviderTypes.Huawei)
+                else if (context.Registration.ProviderType is ProviderTypes.Huawei or ProviderTypes.Twitch)
```

Comment:
Note: this code block adds the device authorization flow to the supported grant types, which isn't something we want to do if we're not going to support it 😃

Reply:
Did you have a chance to test the client credentials grant?

Reply:
Fixed it.

Yes, tested for:

- `Client credentials grant (application authentication only)`
- `Authorization code flow` + `Refresh token`
- `Let OpenIddict negotiate the best authentication flow` + `Refresh token`

Reply:
Excellent, thanks 👍🏻